File: service/src/main/java/com/ricardococati/service/converter/CandlestickMessageConverter.java
package com.ricardococati.service.converter;

import static java.util.Objects.nonNull;

import com.ricardococati.model.entities.Candlestick;
import com.ricardococati.model.entities.CandlestickDiario;
import com.ricardococati.model.entities.CandlestickDiarioMessage;
import com.ricardococati.model.entities.CandlestickSemanal;
import com.ricardococati.model.entities.CandlestickSemanalMessage;
import java.time.LocalDate;
import org.springframework.stereotype.Component;

@Component
public class CandlestickMessageConverter {

  private static final boolean DEFAULT_VALUE = false;

  public CandlestickDiario convert(CandlestickDiarioMessage message) {
    return CandlestickDiario
        .builder()
        .dtpreg(convertStringToLocalDate(message.getDtpreg()))
        .candlestick(
            Candlestick
                .builder()
                .codneg(message.getCodneg())
                .preabe(message.getPreabe())
                .preult(message.getPreult())
                .premin(message.getPremin())
                .premax(message.getPremax())
                .voltot(message.getVoltot())
                .semana(message.getSemana())
                .build())
        .build();
  }

  private LocalDate convertStringToLocalDate(final String date) {
    LocalDate parse = LocalDate.now();
    if (nonNull(date)) {
      parse = LocalDate.parse(date);
    }
    return parse;
  }

  public CandlestickSemanal convertSemanal(CandlestickSemanalMessage message) {
    return CandlestickSemanal
        .builder()
        .dtpregini(convertStringToLocalDate(message.getDtpregini()))
        .dtpregfim(convertStringToLocalDate(message.getDtpregfim()))
        .candlestick(
            Candlestick
                .builder()
                .codneg(message.getCodneg())
                .preabe(message.getPreabe())
                .preult(message.getPreult())
                .premin(message.getPremin())
                .premax(message.getPremax())
                .voltot(message.getVoltot())
                .semana(message.getSemana())
                .build())
        .build();
  }

}

File: service/src/main/java/com/ricardococati/service/converter/CandlestickMessageConverter.kt
package com.ricardococati.service.converter

import com.ricardococati.model.entities.Candlestick
import com.ricardococati.model.entities.CandlestickDiario
import com.ricardococati.model.entities.CandlestickDiarioMessage
import com.ricardococati.model.entities.CandlestickSemanal
import com.ricardococati.model.entities.CandlestickSemanalMessage
import org.springframework.stereotype.Component
import java.time.LocalDate
import java.util.Objects

@Component
class CandlestickMessageConverter {

    fun convert(message: CandlestickDiarioMessage): CandlestickDiario {
        return CandlestickDiario
                .builder()
                .dtpreg(convertStringToLocalDate(message.dtpreg))
                .candlestick(
                        Candlestick
                                .builder()
                                .codneg(message.codneg)
                                .preabe(message.preabe)
                                .preult(message.preult)
                                .premin(message.premin)
                                .premax(message.premax)
                                .voltot(message.voltot)
                                .semana(message.semana)
                                .build())
                .build()
    }

    private fun convertStringToLocalDate(date: String?): LocalDate {
        var parse = LocalDate.now()
        if (Objects.nonNull(date)) {
            parse = LocalDate.parse(date)
        }
        return parse
    }

    fun convertSemanal(message: CandlestickSemanalMessage): CandlestickSemanal {
        return CandlestickSemanal
                .builder()
                .dtpregini(convertStringToLocalDate(message.dtpregini))
                .dtpregfim(convertStringToLocalDate(message.dtpregfim))
                .candlestick(
                        Candlestick
                                .builder()
                                .codneg(message.codneg)
                                .preabe(message.preabe)
                                .preult(message.preult)
                                .premin(message.premin)
                                .premax(message.premax)
                                .voltot(message.voltot)
                                .semana(message.semana)
                                .build())
                .build()
    }

    companion object {
        private const val DEFAULT_VALUE = false
    }

}